rt/vm/src/main/java/org/apidesign/vm4brwsr/Main.java

```diff
@@ -43,6 +43,7 @@ private Main() {}
     public static void main(String... args) throws IOException, URISyntaxException {
         final String obfuscate = "--obfuscatelevel";
         final String extension = "--createextension";
+        final String sourcemap = "--sourcemap";
```

Comment:
I have forgotten there is a `main` method in the JAR! I assume you are using this `main` for testing, right? What are the CLI commands you use?

It'd be good to integrate (or describe how such integration should look like) with Maven and test infrastructure. 

There are two versions when one generates `-bck2brwsr.jar` artifacts with pre-compiled Bck2Brwsr `.js` files - production (usually obfuscated) and debugging one. It'd be great if the source maps (and sources?) could be attached to the debug one.

Then there is the `VMTest` functionality - it should generate source maps by default and make them available when the test is running in the browser.

Reply:
Heuréka. I've got it working with:
```
bck2brwsr$ export R=`pwd`
bck2brwsr/javaquery/demo-calculator/target/demo.static.calculator-1.0-SNAPSHOT-bck2brwsr/public_html/app$ ln -s ../../../../src/main/java/org/ .
bck2brwsr/javaquery/demo-calculator$ java -ea -cp $R/rt/emul/compact/target/emul-1.0-SNAPSHOT.jar:$R/rt/emul/mini/target/emul.mini-1.0-SNAPSHOT.jar:$R/rt/vm/target/vm4brwsr-1.0-SNAPSHOT.jar:target/demo.static.calculator-1.0-SNAPSHOT.jar org.apidesign.vm4brwsr.Main --createextension --obfuscatelevel NONE --sourcemap target/demo.static.calculator-1.0-SNAPSHOT-bck2brwsr/public_html/app/calculator.js org/apidesign/bck2brwsr/demo/calc/staticcompilation/Preinit 
bck2brwsr/javaquery/demo-calculator$ mvn bck2brwsr:show
```
I am able to regenerate the `calculator.js` and put breakpoint into `Preinit.sayHello` Java source code. The chrome debugger stops there and shows value of `msg` parameter. Great work!

Reply:
Please accept this https://github.com/jtulach/bck2brwsr/commit/d027e0089f309e265d925772a7f27934fde15e1c into your PR. It is adding `Bck2Brwsr.generate(Appendable, String, Appendable)`. Next I plan to use it from Maven plugin.

Reply:
With https://github.com/jtulach/bck2brwsr/commit/1ddac1ef970750f48b7ceb2e96e18c563c132142 I can build `bck2brwsr` and then:
```
bck2brwsr$ mvn -f javaquery/demo-calculator/ clean package bck2brwsr:show \
    -Dbck2brwsr.obfuscationlevel=NONE -DskipTests
```
the calculator application is shown and debugging works like a charm!

![image](https://user-images.githubusercontent.com/1842422/178131012-5b620853-a17e-48db-8d71-7207fcbf17e4.png)